datagrid_gtk3/utils/transformations.py

```diff
@@ -448,5 +449,9 @@ def degree_decimal_str_transform(value, length=8):
     if len(value) < length:
         value = '0' * (length - len(value)) + value
 
+    # Add decimal point at the expected location
     value = '{}.{}'.format(value[:2], value[2:])
-    return float(value)
+
+    # Remove non-significant leading zeroes
+    value = re.sub(r'0+(\d\.)', r'\1', value)
```

Comment:
A suggestion (you might prefer your solution):

Why not do:

```python
value = decimal.Decimal('{}.{}'.format(value[:2], value[2:]))
return str(value)
```

The `decimal.Decimal` should take care of removing any non-significant leading zeroes and it will not loose precision. It will also raise an exception in case `value` is not a valid decimal number, which might prevent some errors.